backend/de.metas.adempiere.adempiere/migration/src/main/sql/postgresql/system/10-de.metas.adempiere/5656080_sys_gh107_isDunning_Default.sql

```diff
@@ -0,0 +1,115 @@
+INSERT INTO AD_Index_Table (AD_Client_ID,AD_Index_Table_ID,AD_Org_ID,AD_Table_ID,BeforeChangeCode,BeforeChangeCodeType,Created,CreatedBy,EntityType,ErrorMsg,IsActive,IsUnique,Name,Processing,Updated,UpdatedBy,WhereClause) VALUES (0,540707,0,114,'ab182_isDunningDefault=''N''','SQLS',TO_TIMESTAMP('2022-09-14 11:02:50','YYYY-MM-DD HH24:MI:SS'),100,'D','Only one dunning default contact can be active. If you change it, the previous dunning default contact will be automatically unchecked.','Y','Y','AD_User_isDunningDefault','N',TO_TIMESTAMP('2022-09-14 11:02:50','YYYY-MM-DD HH24:MI:SS'),100,'C_BPartner_ID IS NOT NULL AND ab182_isDunningDefault=''Y''')
+;
+
+-- 2022-09-14T09:02:51.051Z
+INSERT INTO AD_Index_Table_Trl (AD_Language,AD_Index_Table_ID, ErrorMsg, IsTranslated,AD_Client_ID,AD_Org_ID,Created,Createdby,Updated,UpdatedBy,IsActive) SELECT l.AD_Language, t.AD_Index_Table_ID, t.ErrorMsg, 'N',t.AD_Client_ID,t.AD_Org_ID,t.Created,t.Createdby,t.Updated,t.UpdatedBy,'Y' FROM AD_Language l, AD_Index_Table t WHERE l.IsActive='Y'AND (l.IsSystemLanguage='Y') AND t.AD_Index_Table_ID=540707 AND NOT EXISTS (SELECT 1 FROM AD_Index_Table_Trl tt WHERE tt.AD_Language=l.AD_Language AND tt.AD_Index_Table_ID=t.AD_Index_Table_ID)
+;
+
+-- 2022-09-14T09:03:14.075Z
+INSERT INTO AD_Index_Column (AD_Client_ID,AD_Column_ID,AD_Index_Column_ID,AD_Index_Table_ID,AD_Org_ID,Created,CreatedBy,EntityType,IsActive,SeqNo,Updated,UpdatedBy) VALUES (0,5844,541268,540707,0,TO_TIMESTAMP('2022-09-14 11:03:13','YYYY-MM-DD HH24:MI:SS'),100,'U','Y',10,TO_TIMESTAMP('2022-09-14 11:03:13','YYYY-MM-DD HH24:MI:SS'),100)
+;
+
+-- 2022-09-14T09:03:34.531Z
+UPDATE AD_Index_Column SET EntityType='D',Updated=TO_TIMESTAMP('2022-09-14 11:03:34','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Index_Column_ID=541268
+;
+
+-- 2022-09-14T09:03:48.649Z
+INSERT INTO AD_Index_Column (AD_Client_ID,AD_Column_ID,AD_Index_Column_ID,AD_Index_Table_ID,AD_Org_ID,Created,CreatedBy,EntityType,IsActive,SeqNo,Updated,UpdatedBy) VALUES (0,584329,541269,540707,0,TO_TIMESTAMP('2022-09-14 11:03:48','YYYY-MM-DD HH24:MI:SS'),100,'D','Y',20,TO_TIMESTAMP('2022-09-14 11:03:48','YYYY-MM-DD HH24:MI:SS'),100)
+;
+
+-- 2022-09-14T09:09:18.223Z
+UPDATE AD_Index_Table_Trl SET IsTranslated='Y',Updated=TO_TIMESTAMP('2022-09-14 11:09:18','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Index_Table_ID=540707 AND AD_Language='en_US'
+;
+
+
+-- 2022-09-14T09:17:29.304Z
+UPDATE AD_Index_Table SET BeforeChangeWarning='Do you really want to change the dunning default?',Updated=TO_TIMESTAMP('2022-09-14 11:17:29','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Index_Table_ID=540707
+;
+
+-- 2022-09-14T09:17:31.508Z
+DROP INDEX IF EXISTS ad_user_isdunningdefault
+;
+
+-- 2022-09-14T09:17:31.511Z
+CREATE UNIQUE INDEX AD_User_isDunningDefault ON AD_User (C_BPartner_ID,ab182_isDunningDefault) WHERE C_BPartner_ID IS NOT NULL AND ab182_isDunningDefault='Y'
+;
+
+-- 2022-09-14T09:17:31.520Z
+CREATE OR REPLACE FUNCTION AD_User_isDunningDefault_tgfn()
```

Comment:
I would suggest having this as a model interceptor instead and avoid db triggers.

Also, pls rename column name `ab182_isDunningDefault` to `IsDunningDefault`.